### DISCUSSION TITLE
Having issues filtering the products API in a certain way?

### DISCUSSION THREAD

**Discussion by seb86** [co-cart/co-cart#390] [Requests for Comments]
I'm currently reviewing the filtering capabilities and logging what works and does not.

If it is within the possibility of the core of WooCommerce and not an extension, then please comment below what you have struggled with or what you would like as a possibility with the products API.

Please share while I do a validation.

Thank you.

kind regards,

Sébastien.

P.S. Issues I find will be logged here: https://github.com/co-cart/cocart-products-api/issues/15

**Comment by seb86** [comment]
I'm wondering if I need to also filter out variations that don't match the filtered attributes for the returned products or do I leave variables to return all variations?

**Comment by Poylar** [comment]
Are there any notes already on what works and what doesn’t? Since neither woocommerce nor cocart support filtering by several variations, I had to write my own solution, but if there is a solution through cocart, then I would use it.

**Reply by seb86** [reply]
@Poylar Feel free to share how you are wanting to filter variations and after the holidays it will get looked into. Currently updating the query for products so it's utilizing improvements made in WooCommerce, so now would be a good time to provide what you have attempted, why it doesn't work and your use cases.

**Reply by Poylar** [reply]
I would like to be able to filter products into different categories.
For example
/products?pa_size=[1,2,3]&pa_color=[4,5,6]

This way I will get all available products with ID sizes 1,2,3 and ID colors 4,5,6. All other products will be excluded from the response.

You may also need the same thing for categories and tags (but to be honest, I haven’t checked whether sampling from several categories/tags works there now

**Reply by seb86** [reply]
I think you will find that filtering the products via the attribute values instead of ID is how it will work.
